codes/c/chapter_divide_and_conquer/build_tree.c

```diff
@@ -0,0 +1,57 @@
+/**
+  * File : build_tree.c
+  * Created Time: 2023-10-07
+  * Author : lucas (superrat6@gmail.com)
+  */
+
+#include "../utils/common.h"
+
+/* 构建二叉树：分治 */
+TreeNode* dfs(int* preorder, int* inorderMap, int i, int l, int r, int size) {
+    // 子树区间为空时终止
+    if (r - l < 0)
+        return NULL;
+    // 初始化根节点
+    TreeNode* root = (TreeNode*)malloc(sizeof(TreeNode));
+    root->val = preorder[i];
+    root->left = NULL;
+    root->right = NULL;
+    // 查询 m ，从而划分左右子树
+    int m = inorderMap[preorder[i]];
+    // 子问题：构建左子树
+    root->left = dfs(preorder, inorderMap, i + 1, l, m - 1, size);
+    // 子问题：构建右子树
+    root->right = dfs(preorder, inorderMap, i + 1 + m - l, m + 1, r, size);
+    // 返回根节点
+    return root;
+}
+
+/* 构建二叉树 */
+TreeNode* buildTree(int* preorder, int preorderSize, int* inorder, int inorderSize) {
+    // 初始化哈希表，存储 inorder 元素到索引的映射
+    int* inorderMap = (int*)malloc(sizeof(int) * 10);
```

Comment:
assume that the max element is 9?

Recommend using a large number like 1000 and add a comment

Reply:
It can of course be a very large number, and can be traversed to find the largest value and then add 1.